eclass/toolchain-funcs.eclass

```diff
@@ -523,11 +593,12 @@ tc-ld-disable-gold() {
 # @FUNCTION: tc-ld-force-bfd
 # @USAGE: [toolchain prefix]
 # @DESCRIPTION:
-# If the gold or lld linker is currently selected, configure the compilation
-# settings so that we use the bfd linker instead.
+# If a linker other than bfd is currently selected, configure the compilation
+# settings so that we use the bfd linker instead.  This function should not
+# be used for simple underlinking problems.  This function is intended for use
+# when a package is fragile and/or relies on bfd internals.
 tc-ld-force-bfd() {
-	if ! tc-ld-is-gold "$@" && ! tc-ld-is-lld "$@" ; then
-		# They aren't using gold or lld, so nothing to do!
+	if tc-ld-is-bfd "$@" ; then
 		return
 	fi
```

Comment:
Hm, not a request for changes (just commenting) but was thinking about how `tc-ld-force-bfd` should probably never be done without also doing `strip-unsupported-flags` (e.g. fails if you're passing `-flto=thin` and was expecting lld), but then I don't want to do `strip-unsupported-flags` for nothing if linker did not actually change due to edge cases of flags needing to be together to work.

Feel like /all/ ebuild using this should do something like:
```bash
if ! tc-ld-is-bfd; then
	tc-ld-force-bfd
	strip-unsupported-flags
fi
```
+BDEPEND on binutils

Technically duplicates the tc-ld-is-bfd check but guess no biggie.

Other options could be to strip from toolchain-funcs (albeit its been avoiding inheriting flag-o-matic so far), use return values for shorter `tc-ld-force-bfd && strip-unsupported-flags` (but that may be less clear anyway).

Reply:
..then again I guess this is pointless right now, just tried `strip-unsupported-flags` with jansson and it's not removing `-flto=thin` then failing anyway

Guess two issues at a glance:
1. LDFLAGS tests ignores the `-fuse-ld=bfd` that comes after and still tests `-fuse-ld=lld` (takes first rather than latest)
2. it doesn't pass LDFLAGS during the CFLAGS tests, so it's always testing with clang's default (and I use default-lld)
`DEBUG: clang -Werror -flto=thin -xc -c /tmp/portage/dev-libs/jansson-2.14-r1/temp/test-flag.c -o /tmp/portage/dev-libs/jansson-2.14-r1/temp/test-flag.exe`

At least wine is mostly spared given it does a hard `strip-flags` unless `USE=custom-cflags`